sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -5704,10 +5704,9 @@ class SQLConf extends Serializable with Logging with SqlApiConf {
     settings.synchronized { settings.asScala.toMap }
 
   /**
-   * Return all the configuration definitions that have been defined in [[SQLConf]]. Each
-   * definition contains key, defaultValue and doc.
+   * Return all the public configuration definitions that have been defined in [[SQLConf]].
    */
-  def getAllDefinedConfs: Seq[(String, String, String, String)] = {
+  def getAllDefinedConfs: Seq[(String, String, String, String, Set[String])] = {
```

Comment:
I think this is only used by us, and I also think it's the right place to make a change judging from the docstring. We also have a rare chance to "get away" with breaking public APIs due to the upcoming 4.0 release.

But it is indeed a public API. Do you prefer I just create a new API? It will duplicate this existing API but include the new documentation groups as well. I was thinking to also have this return a `Seq` of a case class with descriptive attribute names, so we're not calling `._1` and the like everywhere.